compiler/src/dotty/tools/dotc/cc/CaptureRef.scala

```diff
@@ -135,14 +142,26 @@ trait CaptureRef extends TypeProxy, ValueType:
               case _ => false
           || viaInfo(y.info)(subsumingRefs(this, _))
         case MaybeCapability(y1) => this.stripMaybe.subsumes(y1)
+        case y: TypeRef if y.derivesFrom(defn.Caps_CapSet) =>
+          y.info match
+            case TypeBounds(_, hi: CaptureRef) => this.subsumes(hi)
```

Comment:
Can this case happen? I thought the upper bound has to be of the form `CapSet^cs`? If it can happen, maybe add a comment explaining why.

Reply:
The upper and lower bounds don't have to be in the form of `CapSet^{...}`. They can also be other capture set variables, which are bounded by `CapSet`, like `def test[X^, Y^, Z >: X <: Y]`. With this set up, both cases would happen during `subsumes`.